src/services/prompter/database-prompts.js

```diff
@@ -25,17 +26,17 @@ class DatabasePrompts extends AbstractPrompter {
     this.programArguments = programArguments;
   }
 
-  async handlePrompts() {
+  async handlePrompts(forSql = true, forNosql = true) {
     this.handleConnectionUrl();
-    this.handleDialect();
+    this.handleDialect(forSql, forNosql);
     this.handleName();
     this.handleSchema();
     this.handleHostname();
     this.handlePort();
     this.handleUser();
     this.handlePassword();
     this.handleSsl();
-    this.handleMongodbSrv();
+    if (forNosql) this.handleMongodbSrv();
```

Comment:
We already have this kind of checks. What is requested in prompts is handled by the `requests` object. Why didn't you use it to specify the prompts ?

Reply:
Actually it's not realy handled by the `requests` object here, `mongoDbSrv` is always in the `requests` but on the prompt definition we only ask it if the dialect is mongo (so this if is useless you are right)